src/nvim/compositor.c

```diff
@@ -0,0 +1,369 @@
+// This is an open source non-commercial project. Dear PVS-Studio, please check
```

Comment:
name it `ui_compositor.c` , so it is grouped with `ui.c` and `ui_bridge.c` ? And the prefix could be `uicomp_`. This helps discover the project structure.

Reply:
yeah, lets do that.